src/Sylius/Bundle/CoreBundle/CatalogPromotion/CommandDispatcher/BatchedApplyCatalogPromotionsOnVariantsCommandDispatcher.php

```diff
@@ -18,10 +18,17 @@
 
 final class BatchedApplyCatalogPromotionsOnVariantsCommandDispatcher implements ApplyCatalogPromotionsOnVariantsCommandDispatcherInterface
 {
+    /** @var positive-int */
+    private int $size;
+
     public function __construct(
```

Comment:
Won't this work the same?

```suggestion
    public function __construct(
        private MessageBusInterface $messageBus,
        private int $size,
    ) {
        if ($size < 1) {
            throw new \InvalidArgumentException('Size must be greater than 0');
        }
    }
```